File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums):
        myMap = {}
        startElements = []
        for i, el in enumerate(nums):
            myMap[el] = 1
            if el-1 not in nums:
                startElements.append(el)
        longest = 0
        for startEl in startElements:
            posl = 1
            while True:
                el = myMap.get(startEl+posl, -1)
                if el != -1:
                    posl +=1
                else:
                    if posl > longest:
                        longest = posl
                    break
        return longest

File: Data Structures & Algorithms/two-integer-sum/submission-3.go
func twoSum(nums []int, target int) []int {
	myMap := make(map[int]int)
	for i := 1; i <= len(nums); i++ {
		a := myMap[target-nums[i-1]]
		if a != 0 {
			return []int{a - 1, i - 1}
		}
		myMap[nums[i-1]] = i
	}
	return nil
}
